packages/frontend-2/components/invite/dialog/shared/SelectUsers.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <LayoutDialog
+    v-model:open="isOpen"
+    :buttons="dialogButtons"
+    max-width="md"
+    @update:open="onUpdateOpen"
+  >
+    <template #header>
+      {{ title }}
+    </template>
+    <form @submit="onSubmit">
+      <div class="flex flex-col gap-y-3 text-foreground">
+        <div v-for="(item, index) in fields" :key="item.key" class="flex gap-x-3">
+          <div class="flex flex-col gap-y-3 flex-1">
+            <div class="flex flex-row gap-x-3">
+              <div class="flex-1">
+                <FormTextInput
+                  v-model="item.value.email"
+                  :name="`email-${item.key}`"
+                  color="foundation"
+                  placeholder="Email address"
+                  show-clear
+                  full-width
+                  use-label-in-errors
+                  :show-label="index === 0"
+                  label="Email"
+                  :rules="[
+                    isEmailOrEmpty,
+                    canHaveRole({
+                      allowedDomains: props.allowedDomains,
+                      workspaceRole: item.value.workspaceRole
+                    })
+                  ]"
+                  :help="
+                    item.value.matchesDomainPolicy === false
+                      ? 'This email does not match the set domain policy, and can only be invited as a guest'
+                      : undefined
+                  "
+                />
+              </div>
+              <FormSelectWorkspaceRoles
+                v-model="item.value.workspaceRole"
+                label="Select role"
+                :name="`project-role-${item.key}`"
+                class="sm:w-44"
+                mount-menu-on-body
+                :allow-unset="false"
+                :show-label="index === 0"
+                :disabled-items="getDisabledWorkspaceItems(item.value.email)"
+                disabled-item-tooltip="This email does not match the set domain policy, and can only be invited as a guest"
+              />
+            </div>
+          </div>
+          <div class="relative w-4">
+            <CommonTextLink
```

Comment:
Not too important, but for icon clicks i've been using FormButton with `hide-text` `subtle` and `icon-left`. This means there is a hover effect consistent with our other buttons